pkg/api/addons.go

```diff
@@ -677,6 +678,13 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		},
 	}
 
+	// set host network to true for single stack IPv6 as the the nameserver is currently
+	// IPv4 only. By setting it to host network, we can leverage the host routes to successfully
+	// resolve dns.
+	if cs.Properties.FeatureFlags.IsFeatureEnabled("EnableIPv6Only") {
```

Comment:
We should add UT coverage to `addons_test.go` for these ipv6 scenarios. Lemme know if you want help, it's 4k+ lines of UT coverage :)

Reply:
Let me take a stab at it, will ping you if you I need help :)

Reply:
@jackfrancis added the tests, PTAL!